batch/batch/front_end/front_end.py

```diff
@@ -1489,6 +1499,89 @@ async def insert(tx):
     set_message(session, f'Added billing project {billing_project}.', 'info')
     return web.HTTPFound(deploy_config.external_url('batch', '/billing_projects'))
 
+async def _close_billing_project(db, billing_project):
+    @transaction(db)
+    async def close_project(tx):
+        row = await tx.execute_and_fetchone(
+            '''
+SELECT name, closed,
+  (SELECT id FROM batches WHERE 
+    billing_project = billing_projects.name AND 
+    state != 'complete' LIMIT 1) AS batch
+FROM billing_projects WHERE name = %s FOR UPDATE;
+    ''',
+            (billing_project,))
+        if not row:
+            raise KeyError(billing_project)
+        assert row['name'] == billing_project
+        if row['closed']:
+            raise ValueError(f'Billing project {billing_project} is already closed.')
+        if row['batch'] is not None:
+            raise RuntimeError(f'Billing project {billing_project} has open or running batches.')
+
+        await tx.execute_update(
+            'UPDATE billing_projects SET closed = TRUE WHERE name = %s;',
```

Comment:
I think you can say `closed = 1`. That's what we do in other places for booleans.